packages/webapp/src/hooks/useOfflineCheck.ts

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from "react";
+import { toast } from "sonner";
+
+async function performNetworkCheck() {
+  const endpoint = "https://www.google.com";
+  try {
+    const response = await fetch(endpoint, {
+      method: "HEAD",
+      cache: "no-cache",
+      mode: "no-cors",
+    });
+    return response.ok;
+  } catch (error) {
+    throw error; // Rethrow for retries
+  }
+}
+
+function useOfflineCheck(options: any = {}) {
+  const {
+    // Optional configuration properties:
+    threshold = 1000, // Minimum interval between offline checks (ms)
+    maxRetries = 3, // Maximum number of retries before assuming offline
+    retryDelay = 1500, // Delay between retries (ms)
+  } = options;
+
+  const [isOffline, setIsOffline] = useState(false);
+  const [retries, setRetries] = useState(0);
+
+  useEffect(() => {
+    const handleOnline = () => {
+      setIsOffline(false);
+      setRetries(0); // Reset retries on reconnection
+    };
+
+    const handleOffline = () => {
+      if (retries < maxRetries) {
+        setTimeout(() => {
+          performNetworkCheck()
+            .then(() => handleOnline()) // online
+            .catch(() => {
+              setRetries(retries + 1); // Retry if error occurs
+            });
+        }, retryDelay);
+      } else {
+        setIsOffline(true);
+        toast.error("App is offline. Please check your internet connection.");
```

Comment:
Add an option object here with `{ dismissible: false, duration: Infinity }` which prevents the toast from closing.

Reply:
I'm wondering if we need this at all — can't we just handle the "online" and "offline" events? On "online" do what you do in `handleOnline`, on "offline" do `setIsOffline` and the toast. What's the reasoning here? Is there some documentation around the behaviour and reliability of these events?